lib/ansible/module_utils/satellite.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Luc Stroobant <luc.stroobant@wdc.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+try:
+    import requests
```

Comment:
Just as a heads up.  We will not accept this change switching to `requests`.  `requests` is a blacklisted import and we require use of `open_url`.

I see the conversation below, but that is not sufficient.

Requests has too many versions that may not be compatible, there are likely to be issues with fullfilling pinned dependencies, and this adds an extra non-necessary dependency.

Reply:
In that case the module (and specially the reporting module that we also want to contribute) becomes unusable for our use case. I can't spend on rewriting with the internal Ansible url modules if that bug is not resolved.

Reply:
@sivel Maybe we should have a check for this in Shippable CI, as I see more and more modules using requests.
```
[dag@moria ansible.git]$ grep -rlE '(import requests|from requests)' lib/ansible | wc -l
35
```

Reply:
We do actually, it's done in validate-modules.  However `validate-modules` doesn't do a verification in `module_utils`.  A code-smell may not be bad.  However there is also strange logic for checking this, as a few modules snuck in after the re-combining core and extras into the main repo, when validate-modules had some broken functionality. So we would have to add skip lists.

There are also a few modules which by extension of another dependency use requests, such as some cloud modules, so they import to get access to exceptions.

Reply:
@sivel : I removed the requests dependency and can confirm that with the current ansible-devel the open files and performance issues I had with our reporting module have been resolved.